static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -301,47 +303,102 @@ class SpanTreeModel {
-            isLastSibling: index === lastIndex,
-            continuingTreeDepths: descendantContinuingTreeDepths,
-            hiddenSpanGroups,
-            spanGroups: new Set(childSpanGroup),
-            filterSpans,
-            previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
-            event,
-            isOnlySibling: descendantsSource.length === 1,
-            spanGrouping: shouldGroup
-              ? [...(spanGrouping ?? []), wrappedSpan]
-              : undefined,
-            toggleSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.toggleSpanGroup
-                : toggleSpanGroup
-              : undefined,
-            showSpanGroup: isNotLastSpanOfGroup
-              ? toggleSpanGroup === undefined
-                ? this.showSpanGroup
-                : showSpanGroup
-              : false,
-            addTraceBounds,
-            removeTraceBounds,
-          })
-        );
-
-        acc.previousSiblingEndTimestamp = span.span.timestamp;
+        // Groups less than 5 indicate that the spans should be left ungrouped
+        if (group.length < MIN_SIBLING_GROUP_SIZE) {
+          for (const spanModel of group) {
+            acc.descendants.push(
+              ...spanModel.getSpansList({
+                operationNameFilters,
+                generateBounds,
+                treeDepth: shouldHideSpanOfGroup ? treeDepth : treeDepth + 1,
+                isLastSibling: index === lastIndex,
+                continuingTreeDepths: descendantContinuingTreeDepths,
+                hiddenSpanGroups,
+                spanGroups: new Set(childSpanGroup),
+                filterSpans,
+                previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
+                event,
+                isOnlySibling: descendantsSource.length === 1,
+                spanGrouping: shouldGroup
+                  ? [...(spanGrouping ?? []), wrappedSpan]
+                  : undefined,
+                toggleSpanGroup: isNotLastSpanOfGroup
+                  ? toggleSpanGroup === undefined
+                    ? this.toggleSpanGroup
+                    : toggleSpanGroup
+                  : undefined,
+                showSpanGroup: isNotLastSpanOfGroup
+                  ? toggleSpanGroup === undefined
+                    ? this.showSpanGroup
+                    : showSpanGroup
+                  : false,
+                addTraceBounds,
+                removeTraceBounds,
+              })
+            );
+
+            acc.previousSiblingEndTimestamp = spanModel.span.timestamp;
+          }
+
+          return acc;
+        }
+
+        // NOTE: I am making the assumption here that grouped sibling spans will not have children.
+        // By making this assumption, I can immediately wrap the grouped spans here without having
+        // to recursively traverse them.
+
+        // This may not be the case, and needs to be looked into later
```

Comment:
Yeah I don't think we're considering grouping any nodes of the tree aside from leafs for this first pass, we should consider it in the future though, it can be an iteration.